processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java

```diff
@@ -269,12 +270,22 @@ public int compare(Row left, Row right)
 
   private Ordering<Row> metricOrdering(final String column, final Comparator comparator)
   {
-    return Ordering.from(Comparator.comparing((Row row) -> row.getRaw(column), Comparator.nullsLast(comparator)));
+    if (NullHandling.sqlCompatible()) {
+      return Ordering.from(Comparator.comparing((Row row) -> row.getRaw(column), Comparator.nullsFirst(comparator)));
+    } else {
+      return Ordering.from(Comparator.comparing((Row row) -> row.getRaw(column), Comparator.nullsLast(comparator)));
+    }
   }
 
   private Ordering<Row> dimensionOrdering(final String dimension, final StringComparator comparator)
   {
-    return Ordering.from(Comparator.comparing((Row row) -> row.getDimension(dimension).isEmpty() ? null : row.getDimension(dimension).get(0), Comparator.nullsFirst(comparator)));
+    return Ordering.from(
+        Comparator.comparing(
+            (Row row) -> row.getDimension(dimension).isEmpty()
```

Comment:
Don't call `row.getDimension(dimension)` twice. Please extract a small class, don't inline. It's unreadable.

Reply:
extracted method to avoid calling getDimension twice.